File: src/modifiers/listRowInsets.ts
import { createModifier } from './createModifier';

export interface ListRowInsetsParams {
  top?: number;
  leading?: number;
  bottom?: number;
  trailing?: number;
}

/// SwiftUI `.listRowInsets(_:)`. Applies `EdgeInsets` to a row inside a
/// `List`, overriding the default per-row padding. Omitted edges default
/// to 0.
export function listRowInsets(params: ListRowInsetsParams) {
  return createModifier('listRowInsets', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/ListRowInsets.swift
import SwiftUI

/// SwiftUI `.listRowInsets(_:)`. Overrides the default per-row padding of a
/// `List` row. Omitted edges default to 0.
enum RNWListRowInsetsModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("listRowInsets") { view, params, _ in
            let insets = EdgeInsets(
                top: params.double("top") ?? 0,
                leading: params.double("leading") ?? 0,
                bottom: params.double("bottom") ?? 0,
                trailing: params.double("trailing") ?? 0
            )
            return AnyView(view.listRowInsets(insets))
        }
    }
}
